internal/pkg/agent/application/actions/handlers/handler_action_unenroll.go

```diff
@@ -60,11 +73,48 @@ func (h *Unenroll) Handle(ctx context.Context, a fleetapi.Action, acker acker.Ac
 		return fmt.Errorf("invalid type, expected ActionUnenroll and received %T", a)
 	}
 
+	// Find inputs that want to receive UNENROLL action
+	// Endpoint needs to receive a signed UNENROLL action in order to be able to uncontain itself
+	state := h.coord.State()
+	comps, units := findMatchingUnitsByActionType(state, a.Type())
+	if len(comps) > 0 {
+		// Iterate through found components and forward the UNENROLL action
+		for i, comp := range comps {
```

Comment:
Should this be done in parallel? Seems like it would be better and quicker to send the action to each component at the same time.